core/chain_indexer.go

```diff
@@ -473,3 +512,121 @@ func (c *ChainIndexer) removeSectionHead(section uint64) {
 
 	c.indexDb.Delete(append([]byte("shead"), data[:]...))
 }
+
+func (c *ChainIndexer) addOutpointsToIndexer(addressOutpoints map[string][]*types.OutpointAndDenomination, nodeCtx int, config params.ChainConfig, block *types.WorkObject) {
+	utxos := block.QiTransactions()
+
+	for _, tx := range utxos {
+		for _, in := range tx.TxIn() {
+
+			if types.IsCoinBaseTx(tx, block.ParentHash(nodeCtx), config.Location) {
+				// skip coinbase
+				continue
```

Comment:
Why do you skip coinbases? Coinbase outpoints need to be indexed too.

Reply:
He skipped the inputs of the coinbase tx not the outputs. From my understanding, he removes the utxos associated with new inputs.